xdsl/ir/core.py

```diff
@@ -720,20 +721,22 @@ def owner(self) -> Operation | Block:
 
 
 @dataclass(init=False)
-class IRNode(ABC):
+class _IRNode(ABC):
     def is_ancestor(self, op: IRNode) -> bool:
         "Returns true if the IRNode is an ancestor of another IRNode."
```

Comment:
If we're doing drive by fixes can we improve these docstrings? From reading this one I'd expect the roles of `op` and `self` to be flipped